src/middleware/devtools.ts

```diff
@@ -241,7 +318,34 @@ const devtoolsImpl: DevtoolsImpl =
             message.payload,
             (action) => {
               if (action.type === '__setState') {
-                setStateFromDevtools(action.state as PartialState)
+                if (store === undefined) {
+                  setStateFromDevtools(action.state as PartialState)
+                  return
+                }
+                if (
+                  Object.keys(action.state as S).length > 1 ||
+                  Object.keys(action.state as S).length < 1
+                ) {
+                  console.error(
+                    `
+                    [zustand devtools middleware] Unsupported __setState action format. 
+                    When using 'store' option in devtools(), the 'state' should have only one key, which is a value of 'store' that was passed in devtools(),
+                    and value of this only key should be a state object. Example: { "type": "__setState", "state": { "abc123Store": { "foo": "bar" } } }
+                    `
+                  )
+                }
+                if ((action.state as S)[store] === undefined) {
+                  return
+                }
+                const stateFromDevtools = (action.state as S)[store]
+                if (stateFromDevtools) {
+                  if (
+                    JSON.stringify(api.getState()) !==
+                    JSON.stringify(stateFromDevtools)
```

Comment:
This seem pretty unfortunate. Wonder if we have any other solution than deep equal. Should be fine for this PR.

Reply:
Agreed. If there's a better way, it's probably in the docs for the redux devtools somewhere.